pkgs/development/tools/continuous-integration/concourse/assets/main/default.nix

```diff
@@ -1,36 +1,71 @@
 {
   cacert,
+  callPackage,
+  elmPackages,
   gmp,
-  mkYarnPackage,
+  gnused,
+  lib,
   nodePackages,
   nss,
   patchelf,
+  pkgs,
   src,
   stdenv,
   zlib,
 }:
-mkYarnPackage {
-  name = "concourse-main-assets";
-  nativeBuildInputs = [ nodePackages.yarn patchelf cacert ];
-  buildInputs = [ cacert ];
-  inherit src;
-  yarnNix = ./yarn.nix;
-  pkgConfig = {
-    elm = {
-      postInstall = ''
-        find . -type f -executable -exec patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" {} \;
-        find . -type f -executable -exec patchelf --set-rpath "${gmp}/lib:${zlib}/lib:${nss}/lib" {} \;
+let
+  mkElmPackage =
```

Comment:
did you mention this came from hercules ci? If so wouldn't it be better to import it instead of copying it in?

Reply:
True. I will look it up.

There is also a defect in their Nix expression, too. The original `mkElmPackage` function always compiles `.elm` files to `.html`, instead of `.js` file. Compiling to `.html` file with `elm` gives you a HTML page with the Javascript code embedded, which is not what `concourse` is expecting.

Reply:
Nope, their expression is just a demo. https://github.com/hercules-ci/elm2nix/blob/90ea8c76735de0c80aed832d78e551f2afa38dde/data/default.nix#L1